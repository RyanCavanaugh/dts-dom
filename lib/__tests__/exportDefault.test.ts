import {ContextFlags, create, emit} from "../index"

describe("an export default assignment", () => {
    it("matches the snapshot", () => {
        const exportDefault = create.exportDefault('test');
        expect(emit(exportDefault)).toMatchSnapshot();
    });

    describe("as module member", () => {
        it("matches the snapshot", () => {
            const module = create.module('my-module');
            const constDeclaration = create.const('test', 'string');
            const exportDefault = create.exportDefault('test');
            module.members.push(constDeclaration, exportDefault);
            expect(emit(module)).toMatchSnapshot();
        });
    });
});
