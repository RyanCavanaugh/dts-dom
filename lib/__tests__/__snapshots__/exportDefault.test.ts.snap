// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an export default assignment as module member matches the snapshot 1`] = `
declare module 'my-module' {
    const test: string;

    export default test;

}


`;

exports[`an export default assignment matches the snapshot 1`] = `
export default test;


`;
